drivers/include/periph/hwcrypto.h

```diff
@@ -0,0 +1,401 @@
+ */
+bool hwcrypto_hash_supported(hwcrypto_t dev, hwcrypto_hash_t hash);
+
+/**
+ * @brief   Initialize the hardware crypto peripheral.
+ *
+ * The method should initialize the peripheral. It may power on the device if
+ * needed, but it must ensure that the peripheral is powered off when exiting
+ * this method.
+ *
+ * This function is intended to be called by the board initialization code
+ * during system startup to prepare the (shared) hardware crypto device for
+ * further usage. It uses the board specific initialization parameters as
+ * defined in the board's `periph_conf.h`.
+ *
+ * Errors (e.g. invalid @p dev parameter) are not signaled through a return
+ * value, but should be signaled using the assert() function internally.
+ *
+ * @param[in] dev           the device to initialize
+ */
+void hwcrypto_init(hwcrypto_t dev);
+
+/**
+ * @brief   Initialize hardware crypto device for a given encryption or
+ *          decryption algorithm.
+ *
+ * The mode parameter can be used to change the cipher's mode of operation. If
+ * a mode does not apply, use @p HWCRYPTO_MODE_NONE.
+ *
+ * This operation may alter the peripheral's state, therefore it should be used
+ * exclusively.
+ *
+ * @param[in] dev           the device to initialize
+ * @param[in] cipher        cipher to initialize
+ * @param[in] mode          cipher mode of operation
+ *
+ * @return                  HWCRYPTO_OK on success
+ * @return                  HWCRYPTO_NOTSUP if cipher or mode is not supported
+ * @return                  HWCRYPTO_ERROR if other error
+ */
+int hwcrypto_cipher_init(hwcrypto_t dev,
+                         hwcrypto_cipher_t cipher,
+                         hwcrypto_mode_t mode);
+
+/**
+ * @brief   Set an option for the given hardware crypto device in cipher mode.
+ *
+ * If the given option or value is not supported by the cipher, an error code
+ * is returned.
+ *
+ * This operation may alter the peripheral's state, therefore it should be used
+ * exclusively.
+ *
+ * A hardware crypto device not initialized for cipher mode may produce
+ * unexpected results.
+ *
+ * The peripheral's state may be undefined if this method is used after one or
+ * more encryption or decryption round(s).
+ *
+ * @param[in] dev           the device to initialize
+ * @param[in] option        option from @p hwcrypto_opt_t to initialize
+ * @param[in] value         pointer to option value
+ * @param[in] size          option size
+ *
+ * @return                  HWCRYPTO_OK on success
+ * @return                  HWCRYPTO_NOTSUP if option not supported or
+ *                          applicable
+ * @return                  HWCRYPTO_INVALID if value is not supported or
+ *                          applicable
+ * @return                  HWCRYPTO_ERROR if other error
+ */
+int hwcrypto_cipher_set(hwcrypto_t dev,
+                        hwcrypto_opt_t option,
+                        const void* value,
+                        uint32_t size);
```

Comment:
Why not having separate functions for distinct operations, e.g.?

1. `int hwcrypto_cipher_key(hwcrypto_t dev, const void *key, const void *iv);`
2. `int hwcrypto_cipher_padding(hwcrypto_t dev, uint8_t padding);`
3. `int hwcrypto_cipher_counter(hwcrypto_t dev, uint64_t nonce, uint64_t counter);`